src/sparseml/pytorch/utils/mfac_helpers.py

```diff
@@ -38,6 +50,75 @@
 ]
 
 
+class GradSampler:
+    """
+    Class for computing gradient samples for a Model given a sample data loader and
+    loss function.
+
+    :param dataloader: iterator of data samples to use as model inputs and their loss
+        targets. Samples can either be single tensors as model input or a list of
+        inputs and should be iterated in tuples with their targets
+    :param loss_fn: function to be called on model outputs to compute the loss at
+        each step
+    """
+
+    def __init__(
+        self,
+        dataloader: Iterator[Tuple[Union[Tensor, List[Tensor]], Any]],
+        loss_fn: Callable[[Tensor], Tensor],
+    ):
+        self._dataloader = dataloader
+        self._loss_fn = loss_fn
+
+    @staticmethod
```

Comment:
static method won't allow the implementer to override -- should probably change out for class or instance if we expect there's any chance they'll need to access state